page.go

```diff
@@ -48,8 +54,40 @@ type Page interface {
 	Emoji() string
 }
 
+type Date struct {
+	time.Time
+}
+
+func (t *Date) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var buf string
+	err := unmarshal(&buf)
+	if err != nil {
+		return err
+	}
+
+	var formats = []string{time.DateOnly, time.DateTime}
+	for _, format := range formats {
+		pt, err := time.Parse(format, buf)
+		if err == nil {
+			t.Time = pt
+			break
+		}
+	}
+
+	return nil
+}
+
+type Metadata struct {
+	checked int
+	Title   string   `yaml:"title"`
+	Tags    []string `yaml:"tags"`
+	// supports formats: time.DateOnly and time.DateTime
+	Date Date `yaml:"date"`
```

Comment:
there is no need for this comment I believe. probably we can add a page to docs describing the feature